### PR TITLE
[match][sigh] prefer default keychain during wwdr cert installation

### PR DESCRIPTION
**Pull request by DuMaM**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
When there are multiple keychains WWDR cert is installed in random place.
During search it takes first output from `list-keychains` instead of default one.
https://developer.apple.com/support/certificates/
https://developer.apple.com/support/expiration/

### Description
* expired cert got removed to do not introduce conflicts
* latest cert was added 
* changed priority so default keychain is taken as first

### Testing Steps
1. Create new keychain without any cert
2. Run with `--verbose` lane where `match` install Provision Profile and it's certs. 
3. Take a look on debug log to see if it's using correct keychain for installation WWDR cert.

### CODE REVIEW

File: fastlane_core/lib/fastlane_core/cert_checker.rb
Comment (getaaron) [inline fastlane_core/lib/fastlane_core/cert_checker.rb:93]: @joshdholtz I think removing this should be OK. According to [Apple Worldwide Developer Relations Intermediate Certificate Expiration](https://developer.apple.com/support/expiration/) nobody should be using this any more and it's set to expire on February 7, 2023 anyway, so not much point in installing it.

File: fastlane_core/lib/fastlane_core/cert_checker.rb
Comment (getaaron) [inline fastlane_core/lib/fastlane_core/cert_checker.rb:127]: I'm not sure about this change, according to https://github.com/fastlane-old/fastlane_core/pull/116 we should start with the login keychain - if we want to move default up, maybe we should insert `security login-keychain` at the very top?

File: fastlane_core/spec/cert_checker_spec.rb
Comment (getaaron) [inline fastlane_core/spec/cert_checker_spec.rb:22]:
```suggestion
      it 'should install only the latest official WWDR certificate' do
```

File: fastlane_core/lib/fastlane_core/cert_checker.rb
Comment (getaaron) [inline fastlane_core/lib/fastlane_core/cert_checker.rb:92]: since this only installs one now, maybe we should rename the method
```suggestion
    def self.install_wwdr_certificate
```

File: fastlane_core/lib/fastlane_core/cert_checker.rb
Comment (DuMaM) [reply fastlane_core/lib/fastlane_core/cert_checker.rb:127]: Login keychain is default, most of the time. Also if you have keychain named like aaaa.keychain-db then login one will be not picked here at all. 
From me it has more sense to start default one, because in most cases its used by match and other tools.

File: fastlane_core/lib/fastlane_core/cert_checker.rb
Comment (DuMaM) [reply fastlane_core/lib/fastlane_core/cert_checker.rb:92]: Ok, agree

File: fastlane_core/lib/fastlane_core/cert_checker.rb
Comment (DuMaM) [reply fastlane_core/lib/fastlane_core/cert_checker.rb:93]: Yep. Thats the reason i removed it.
This old cert also was source of many resign problems.

File: fastlane_core/spec/cert_checker_spec.rb
Comment (DuMaM) [inline fastlane_core/spec/cert_checker_spec.rb:26]: I'm not sure it this tests is needed now

**Comment by DuMaM** [comment]
@getaaron could you please review this once more? :)

**Comment by DuMaM** [comment]
It's ready to merge.

File: fastlane_core/lib/fastlane_core/cert_checker.rb
Comment (DuMaM) [reply fastlane_core/lib/fastlane_core/cert_checker.rb:124]: @getaaron 
This old setup is based on this comment
https://github.com/fastlane-old/fastlane_core/pull/116#issuecomment-190934610
Which is not true. Default-keychain is one set by user from gui or `security default-keychain -s <keychain_name>`.
There are chances that it's not set at all, but then we will pick any from list-keychains.

**Comment by getaaron** [comment]
@DuMaM same as your other PR, can you please connect CircleCI to github and ensure all the builds run?

**Review by getaaron — APPROVED** [review]
this looks good to me, approved to merge pending CI, and I'll try to get a second review just in case

**Comment by DuMaM** [comment]
@getaaron CI is fixed here also :)

**Comment by getaaron** [comment]
thanks for the PR and the flaky test fix :)

**Comment by DuMaM** [comment]
Great news 🎉 🦖 
Thanks for merge and have a nice day 😄

**Comment by DuMaM** [comment]
@getaaron I don't know why but ubuntu run failed 😱

**Comment by DuMaM** [comment]
For me it looks like this second ulock_keychain is failing because we got an race condition here.

**Comment by getaaron** [comment]
@DuMaM I tried rerunning the build on the main branch so we can check if it fails consistently https://app.circleci.com/pipelines/github/fastlane/fastlane?branch=master

**Comment by DuMaM** [comment]
@getaaron It another flaky test. I guess they was not designed to run i parallel.
https://github.com/fastlane/fastlane/pull/20448-> i think this will helps

**Comment by grey442** [comment]
I've got a problem with that changes

My case: Two different machines which have installed different wwdr certificate, but have same name "Apple Worldwide Developer Relations Certification Authority". As consuqunce sharing profiles works only on one of two machine.

That code `FastlaneCore::CertChecker.install_wwdr_certificate` checks only name installed certifacates. Why don't add checking expiration date? Because `FastlaneCore::CertChecker.install_wwdr_certificate` install explicit certificate "https://www.apple.com/certificateauthority/AppleWWDRCAG6.cer". 

For example:
`    def self.wwdr_certificate_installed?`
  `      certificate_name = "Apple Worldwide Developer Relations Certification Authority"`
  `      keychain = wwdr_keychain`
  `      date = "Not After : Mar 19 00:00:00 2036 GMT"`
  `      response = Helper.backticks("security find-certificate -a -c '#{certificate_name}' -p #{keychain.shellescape} | openssl x509 -text | grep '#{date}'", print: FastlaneCore::Globals.verbose?)`
  `      certs = response.split("\n")`
  `      certs.count >= 1`
`    end`

**Comment by getaaron** [comment]
Good point, could you open a PR with this change?

**Pull request merged**
